src/routes/lend.js

```diff
@@ -1,80 +1,75 @@
-import React from 'react';
-import { Alert } from 'react-bootstrap';
+import React, { useState, useEffect, useRef } from 'react';
+import { Alert, Container } from 'react-bootstrap';
 import NFTCardGrid from '../components/NFTCardGrid';
 
-class LendPage extends React.Component {
-    constructor(props) {
-        super(props);
-        // TODO: pass in the wallet address from props.
-        this.walletAddress = "0x1086A7DC518546bb8615Df03F23A27433a5EeeE5";
-        this.state = {
-            error: null,
-            owned: [],
-        }
-    }
+function LendPage() {
+    const didFetchNFTsInUserWalletRef = useRef(false);
+    const [nftsInUserWallet, setNFTsInUserWallet] = useState([]);
+    const [nftsListedForLending, setNFTsListedForLending] = useState([]);
+    const [nftsLentOut, setNFTsLentOut] = useState([]);
+    const [error, setError] = useState();
```

Comment:
Is this the new recommended style?
I only went through the basic React tutorial and this seems to be a bit advanced :)

Reply:
Yep, React is moving towards functional components more especially after they released hooks (https://reactjs.org/docs/hooks-overview.html). Previously, if you wanted to add state to components, you'd have to convert to convert to a class component, but now you can do everything you need using functional components.